File: devops-scm-spring-boot-starter/src/main/java/com/tencent/devops/scm/spring/properties/Oauth2ClientProperties.java
package com.tencent.devops.scm.spring.properties;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * oauth2 属性
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Oauth2ClientProperties {
    /**
     * 服务端web url,用于oauth2时页面跳转
     */
    private String webUrl;

    /**
     * 应用ID
     */
    private String clientId;
    /**
     * 应用密钥
     */
    private String clientSecret;
    /**
     * 授权回调url
     */
    private String redirectUri;
}

File: devops-scm-spring-boot-starter/src/main/kotlin/com/tencent/devops/scm/spring/properties/Oauth2ClientProperties.kt
package com.tencent.devops.scm.spring.properties

/**
 * oauth2 属性
 */
data class Oauth2ClientProperties(
    /**
     * 服务端web url,用于oauth2时页面跳转
     */
    var webUrl: String? = null,
    
    /**
     * 应用ID
     */
    var clientId: String? = null,
    
    /**
     * 应用密钥
     */
    var clientSecret: String? = null,
    
    /**
     * 授权回调url
     */
    var redirectUri: String? = null
)